shared/folders/list.desktop.js

```diff
@@ -29,55 +22,56 @@ const Ignored = ({rows, showIgnored, styles, onToggle, isPublic, onClick}) => {
             Ignored folders won't show up on your computer and you won't receive alerts about them.
           </Text>
         </Box>}
-      {showIgnored && rows}
     </Box>
   )
 }
 
-const Rows = ({
-  tlfs = [],
-  installed,
-  isIgnored,
-  type,
-  onOpen,
-  onChat,
-  onClick,
-  onRekey,
-  smallMode,
-}: Props & {isIgnored: boolean, smallMode?: boolean, tlfs?: Array<Folder>}) => (
-  <Box>
-    {!!tlfs &&
-      tlfs.map(tlf => (
-        <Row
-          {...tlf}
-          key={rowKey(tlf.users)}
-          isPublic={type === 'public'}
-          isTeam={type === 'team'}
-          hasReadOnlyUsers={tlf.users && some(tlf.users, 'readOnly')}
-          ignored={isIgnored}
-          installed={installed}
-          onChat={onChat}
-          onClick={onClick}
-          onRekey={onRekey}
-          onOpen={onOpen}
-          smallMode={smallMode}
-        />
-      ))}
-  </Box>
-)
-
 class ListRender extends Component<Props> {
-  _scrollContainer = null
+  static defaultProps: {
+    extraRows: Array<any>,
+    ignored: Array<any>,
+    tlfs: Array<any>,
+  }
+  _renderItem = (index: number, item: any) => {
+    if (index < this.props.extraRows.length) {
+      return this.props.extraRows[index]
+    }
 
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.showIgnored !== this.props.showIgnored && this._scrollContainer && this.props.showIgnored) {
-      // $FlowIssue
-      ReactDOM.findDOMNode(this._scrollContainer).scrollTop += 100
+    // TODO how this works is HORRIBLE. This is just a short term fix to
+    // take this very very old code that we're about to throw out and have it so its not rendering every single row
+    const tlfsIdx = index - this.props.extraRows.length
```

Comment:
this is just a short term fix. we use a react-list and this _renderItem function is rendering 
1. Extra rows (used in widget for the add new folder row)
1. Regular tlfs
1. The ignore divider (if there are ignores)
1. Ignored folders